pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/FileUploadPathProvider.java

```diff
@@ -15,66 +15,105 @@
  */
 package com.linkedin.pinot.controller.api.resources;
 
-import java.io.File;
-import org.apache.commons.io.FileUtils;
 import com.linkedin.pinot.controller.ControllerConf;
+import com.linkedin.pinot.filesystem.PinotFS;
+import com.linkedin.pinot.filesystem.PinotFSFactory;
+import java.io.File;
+import java.net.URI;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 // TODO This is a misc class used by many jersey apis. Need to rename it correctly
 public class FileUploadPathProvider {
   public final static String STATE = "state";
   public final static String TABLE_NAME = "tableName";
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileUploadPathProvider.class);
 
   private final ControllerConf _controllerConf;
 
-  private final File _fileUploadTmpDir;
-  private final File _baseDataDir;
-  private final File _tmpUntarredPath;
-  private final File _schemasTmpDir;
+  private final URI _fileUploadTmpDirURI;
+  private final URI _baseDataDirURI;
+  private final URI _tmpUntarredPathURI;
+  private final URI _schemasTmpDirURI;
   private final String _vip;
 
   public FileUploadPathProvider(ControllerConf controllerConf) throws InvalidControllerConfigException {
     _controllerConf = controllerConf;
+
+    // Pick the correct scheme for PinotFS. For pluggable storage, we will expect that the directory is configured
+    // with a storage prefix. For backwards compatibility, we will assume local/mounted NFS storage if nothing is configured.
+    String scheme = "file";
+    String dataDir = _controllerConf.getDataDir();
+    try {
+      URI uri = new URI(_controllerConf.getDataDir());
+      scheme = uri.getScheme();
+    } catch (Exception e) {
+      // Assume local FS
+      LOGGER.warn("Could not create controller config data dir URI with {}", _controllerConf.getDataDir());
+      dataDir = "file://" + _controllerConf.getDataDir();
+    }
+
+    PinotFS pinotFS = PinotFSFactory.create(scheme);
+
     try {
-      _baseDataDir = new File(_controllerConf.getDataDir());
-      if (!_baseDataDir.exists()) {
-        FileUtils.forceMkdir(_baseDataDir);
+      _baseDataDirURI = new URI(dataDir);
+      if (!pinotFS.exists(_baseDataDirURI)) {
+        pinotFS.mkdir(_baseDataDirURI);
       }
-      _fileUploadTmpDir = new File(_baseDataDir, "fileUploadTemp");
-      if (!_fileUploadTmpDir.exists()) {
-        FileUtils.forceMkdir(_fileUploadTmpDir);
+      _fileUploadTmpDirURI = new URI(_baseDataDirURI + "/fileUploadTemp");
```

Comment:
Let's use private static finals in all the places to make it more readable. (I understand some of them are pre-existing, but good to cleanup).

Reply:
some of these are used outside this class in our apis